Sources/Datadog/Logger.swift

```diff
@@ -203,25 +214,34 @@ public class Logger {
     ///
     /// - Parameter tag: value of the tag
     public func add(tag: String) {
-        loggerTags.insert(tag)
+        queue.async {
+            self.loggerTags.insert(tag)
+        }
     }
 
     /// Removes a tag from all future logs sent by this logger.
     /// Previous logs won't lose the this tag if they were created prior to this call.
     ///
     /// - Parameter tag: value of the tag to remove
     public func remove(tag: String) {
-        loggerTags.remove(tag)
+        queue.async {
+            self.loggerTags.remove(tag)
+        }
     }
 
     // MARK: - Private
 
     private func log(level: LogLevel, message: String, messageAttributes: [String: Encodable]?) {
-        let combinedAttributes = loggerAttributes.merging(messageAttributes ?? [:]) { _, messageAttributeValue in
-            return messageAttributeValue // use message attribute when the same key appears also in logger attributes
+        let combinedAttributes = queue.sync {
```

Comment:
I am thinking here...not sure exactly how the queue works but what if I add a tag and I call `log` after ? The **add-tag** action will be dispatched asynchronously in the queue while the log will be executed synchronously...meaning that you might lose some tags or attributes there right ?  Why not making all "sync" ?

Reply:
let me take some burden off @ncreated's shoulders :) 

by default, `DispatchQueue`s are serial, tasks are scheduled for immediate execution regardless `sync` or `async` is used.

• if we call `sync`, current thread waits until the queue finishes execution
• if we call `async`, current thread continues without waiting for the queue

so, `async` is supposed to give us a performance gain without sacrificing correctness
you also can find more info [here](https://developer.apple.com/documentation/dispatch/dispatchqueue)

Reply:
Tks, like I was suspecting, the DispatchQueue handles everything for you ;)